### PR TITLE
Resolve multiple BB audit issues

### LINKED ISSUE #1474: Remove unused doneBuildingBlock channel
The channel is created in NewManager, then it’s reset every time
in AggregationLoop, but never shared with other parts of the
code.

Remove it

### LINKED ISSUE #1475: Garbage collect Timer in Aggregation Loop
Timer is not stopped when the Aggregation loop exits.

Add a defer timer.Stop statement

### LINKED ISSUE #1476: Context.Done is not honored during chan sends in publishBlock
Honor context.Done to avoid endlessly locking the operation.
This might also create deadlocks.
Select {
Case <-ctx.Done:
Case HeaderCh<-value
}

See BB audit issue:

```
Finding 022 - Context.Done is not honored during chan sends in
publishBlock
```

### LINKED ISSUE #1477: Incomplete Validation of Commit
Commit Validation only asserts if one signature is present, but
does not assert if that signature is valid, or even different from
nil.

Add some minimal validation on the single signature

See BB audit issue:

`Finding 035 - Incomplete Validation of Commit`

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Resolves #1477, resolves #1475, resolves #1474

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved timer management to prevent potential memory leaks.
	- Enhanced error handling during block publishing to handle context cancellation gracefully.
	- Added a validation check for block signatures to ensure data integrity.

### CODE REVIEW

**Review by testinginprod — APPROVED** [review]
LGTM! 👍

**Review by nashqueue — DISMISSED** [review]
ProxyUtAck

**Review by MSevey — COMMENTED** [review]
Can we discuss the select statement changes offline? 
Those are reverting back to conditions that previously caused test failures. 
Let's make sure we aren't re-introducing those or if the other updates, like the shutdown coordination, has resolved the underlying issue.

File: types/block.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ func (c *Commit) ValidateBasic() error {
 	if len(c.Signatures) == 0 {
 		return errors.New("no signatures")
 	}
+	if len(c.Signatures[0]) == 0 {
```
[the remark this answers is not in the document]
Comment (Manav-Aggarwal) [reply types/block.go:105]: Added unit test to capture test case.

**Review by MSevey — APPROVED** [review]

**Pull request merged**